File: src/paths.js
import path from "node:path";

export function memoryHome() {
  return path.resolve(process.env.MEM_HOME || path.join(process.cwd(), ".memoria"));
}

export function dataPaths(base = memoryHome()) {
  return {
    base,
    db: path.join(base, "memory.sqlite"),
    originals: path.join(base, "originals"),
    canonical: path.join(base, "canonical"),
    logs: path.join(base, "logs")
  };
}

File: rust/paths.rs
use std::path::PathBuf;

pub fn memory_home() -> PathBuf {
    std::env::var_os("MEM_HOME")
        .map(PathBuf::from)
        .unwrap_or_else(|| std::env::current_dir().unwrap().join(".memoria"))
}

pub struct DataPaths {
    pub base: PathBuf,
    pub db: PathBuf,
    pub canonical: PathBuf,
    pub originals: PathBuf,
    pub logs: PathBuf,
}

pub fn data_paths(base: Option<PathBuf>) -> DataPaths {
    let base = base.unwrap_or_else(memory_home);
    DataPaths {
        db: base.join("memory.sqlite"),
        canonical: base.join("canonical"),
        originals: base.join("originals"),
        logs: base.join("logs"),
        base,
    }
}
